osquery/core/query.cpp

```diff
@@ -545,11 +569,38 @@ Status serializeQueryData(const QueryData& q,
   return Status();
 }
 
-bool addUniqueRowToQueryData(QueryData& q, const Row& r) {
+bool addUniqueRowToQueryData(QueryDataTyped& q, const RowTyped& r) {
   if (std::find(q.begin(), q.end(), r) != q.end()) {
     return false;
   }
   q.push_back(r);
   return true;
 }
+
+void escapeNonPrintableBytes(std::string& data) {
+  std::string escaped;
+  // clang-format off
+  char const hex_chars[16] = {
+    '0', '1', '2', '3', '4', '5', '6', '7',
+    '8', '9', 'A', 'B', 'C', 'D', 'E', 'F',
+  };
+  // clang-format on
+
+  bool needs_replacement = false;
+  for (size_t i = 0; i < data.length(); i++) {
```

Comment:
const auto &c : data

Reply:
Same as above (I'd rather not modify a piece of code that's been moved in an otherwise unrelated change).

Would be nice if the github diff view tool gave hints about moved chunks of code.